website/docs/features/lightgbm/about.md

```diff
@@ -122,16 +122,57 @@ and can be tuned with [SparkML's cross
 validators](https://spark.apache.org/docs/latest/ml-tuning.html).
 
 Models built can be saved as SparkML pipeline with native LightGBM model
-using `saveNativeModel()`. Additionally, they are fully compatible with [PMML](https://en.wikipedia.org/wiki/Predictive_Model_Markup_Language) and
+using `saveNativeModel()`. Additionally, they're fully compatible with [PMML](https://en.wikipedia.org/wiki/Predictive_Model_Markup_Language) and
 can be converted to PMML format through the
 [JPMML-SparkML-LightGBM](https://github.com/alipay/jpmml-sparkml-lightgbm) plugin.
 
+### Execution Mode
+
+SynapseML must pass data from Spark partitions to LightGBM Datasets before turning over control to
+the native LightGBM execution code. Datasets can either be created per partition (useSingleDatasetMode=false), or
+per executor (useSingleDatasetMode=true). Generally, one Dataset per executor is more efficient since it reduces LightGBM network size and complexity during training or fitting. It also avoids using slow network protocols on partitions
+that are actually on the same executor node.
+
+For the loading of partition data into Datasets, SyanpseML can do it in either a "streaming" or "bulk" execution mode. This mode
+doesn't affect training, just how data is put into Datasets. It can, however, affect memory usage and overall
+fit/transform time. 
+
+#### Bulk Execution mode
+"Bulk" mode is older and requires accumulating all data in executor memory before creating Datasets. This original mode can cause
+OOM errors for large data, especially since the data must be accumulated in its original double-format size.
+For now, "bulk" mode is the default since "streaming" is new, but SynapseML will eventually make streaming the default once
+it has seen more usage.
```

Comment:
```suggestion
The "Bulk" mode requires accumulating all data in executor memory before creating Datasets. This mode can cause
OOM errors for large data, especially since the data must be accumulated in its original double-format size.
For now, "bulk" mode is the default since the "streaming" is new, but SynapseML will eventually make streaming the default.
```